pilot/pkg/model/context.go

```diff
@@ -158,6 +166,49 @@ type Proxy struct {
 
 	// GlobalUnicastIP stores the globacl unicast IP if available, otherwise nil
 	GlobalUnicastIP string
+
+	// Generator is used to generate resources for the node, based on the PushContext.
+	Generator Generator
+
+	// Active contains the list of watched resources for the proxy, keyed by the DiscoveryRequest type.
+	// It is nil if the Proxy uses the default generator
+	Active map[string]*WatchedResource
+}
+
+// WatchedResource tracks an active DiscoveryRequest type.
+type WatchedResource struct {
+	// TypeUrl is copied from the DiscoveryRequest that initiated watching this resource.
+	TypeUrl string
+
+	// Generator is the generator that will generate this resource.
+	Generator Generator
```

Comment:
See comment on the other PR about this. This should probably not be here, or it shouldn't be in proxy. otherwise proxy has a generator and  watched resource has a generator and its confusing what that means.

Reply:
I'll add more comments to address the confusion. 

What it means is that each watched resource can have a specific generator - falling back to the per-proxy generator, and finally to the mesh wide generator ( which currently is the old one)

Reply:
How do we assign per-type generator though? I forgot anther confusion was that generator also takes in a watched resource, and that watched resource has a generator. To me it seems like the proxy should have a generator, and if it needs to do different things for different resources we it can read the watched resource (which is part of the function header) and delegate to whatever it needs. I don't think we need to track this in watched resource itself.

We used to have a lot of these circular references in Pilot which caused a lot of headaches and mostly cleaned them up so if we can avoid more it would be preferred

Reply:
There are several options on how to assign the generator. I don't think they interfere with the interface - or have to be in model. 

I'll remove the Generator from watched resource - and add a comment in Generator that each resource type may have a different generator.